pkg/collector/corechecks/containers/containerd_events.go

```diff
@@ -0,0 +1,294 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build containerd
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	ctrUtil "github.com/DataDog/datadog-agent/pkg/util/containerd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/gogo/protobuf/proto"
+)
+
+//ContainerdEvent contains the timestamp to make sure we flush all events that happened between two checks
+type ContainerdEvent struct {
+	ID        string
+	Timestamp time.Time
+	Topic     string
+	Namespace string
+	Message   string
+	Extra     map[string]string
+}
+
+type Subscriber struct {
```

Comment:
Do we need to export it? Same for `ContainerdEvent`

Reply:
I made it public initially as I envisioned a broader usage from other pkgs, but yeah, it's fine to leave them private for now